### DISCUSSION TITLE
PAT Basic 1033. 旧键盘打字 (20) (C语言实现) | PAT Solution in C

### DISCUSSION THREAD

**Discussion by xlucn** [xlucn/PAT#64] [General]
https://oliverlew.github.io/PAT/Basic/1033.html 

 题目

**Comment by Lemonbillion** [comment]
#include<stdio.h>
int main(void)
{
	static char bk[100004],c;
	static int h;
	static char sk[100005];
	while((c = getchar()) != '\n')
	{
		bk[h++] = c;
	}
	scanf("%s",sk);
	static int i,j,count,upper;
	for(j=0;bk[j] != '\0';j++)
		{
			
			if (bk[j]  == ',' || bk[j]  == '.' || bk[j]  == '-' || bk[j]  == '+')
			{
				upper = 1;
				
			}
		}
	for(i = 0;sk[i] != '\0';i++)
	{
		if (upper)
		{
			if  (sk[i] - 'A' >= 0 && sk[i] - 'Z' <= 0)
			{
				continue;
			 } 
		}
		for(j=0;bk[j] != '\0';j++)
		{
			if (bk[j] == sk[i] || bk[j] - ('A' -'a') == sk[i] || bk[j] + ('A' -'a') == sk[i] ) 
			{
				break;
			}
		}
		if (bk[j] == '\0')
		{
			putchar(sk[i]);
			count++;
		}
	}
	if (!count)
	{
		puts("");
	}
	return 0;
}


测试点4过不了,求助一下
